### PR TITLE
Week 3: Recipe library - Alma and Etna

### PR DESCRIPTION
**Pull request by almaherris**
## Netlify link

https://recipelibrary.netlify.app/

## Collaborators

[Caracal23, almaherris]

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -1,406 +1,399 @@
...
+//Initial rendering of all recipes
+showRecipes(recipes)
+
+//Function random
+const getRandomValue = () => {
+  container.innerHTML = ""
+
+  recipes[Math.floor(Math.random() * recipes.length)]
+  const recipe = recipes[Math.floor(Math.random() * recipes.length)]
+  container.innerHTML += `
```
Comment (Izzibizz) [inline script.js:298]: Here you could have called the showRecipes function and connected it to the result of your getRandomValue function, no need to rewrite the same innerHTML again :)

File: script.js [hunk 2]
```diff
@@ -1,406 +1,399 @@
...
+      <p class="recipe-link"><a href="${
+        recipe.url
+      }" target="_blank">Link to cooking method</a>
+      </div>
+  </div>
+  `
+}
+
+//Function to show favorites
+const showFavorites = () => {
```
Comment (Izzibizz) [inline script.js:320]: Nice to see that you added a favorite function!

File: script.js [hunk 3]
```diff
@@ -1,406 +1,399 @@
...
+  // if selected value = all -> then show all recipes otherwise filter by cuisine type
+  if (selectedCuisine === "all") {
+    showRecipes(recipes)
+  } else {
+    filterRecipesByCuisine(selectedCuisine)
+  }
+})
+
+//Event listener for favorites
+favoriteButton.addEventListener("click", (event) => {
```
Comment (Izzibizz) [inline script.js:371]: Normal buttons have no default behavior, therefore the preventdefault is not necessery for normal buttons, they do not reload the page on default like the submit button would do. Your eventlisteners could be more compact and sleaker by removing this :D

File: script.js [hunk 4]
```diff
@@ -1,406 +1,399 @@
...
+  `
+}
+
+//Function to show favorites
+const showFavorites = () => {
+  const favRecipes = recipes.filter((recipe) => recipe.favorite === true)
+  showRecipes(favRecipes)
+}
+
+//Function to sorting descending
```
Comment (Izzibizz) [inline script.js:325]: I believe that the Ascending and Descending should be reversed when looking at the result on your site

File: script.js [hunk 5]
```diff
@@ -1,406 +1,399 @@
...
+  showRecipes(recipes)
+}
+
+//Function to filter on search input based on name or ingredient
+const filterRecipesBySearchInput = (searchInputValue) => {
+  const filteredRecipes = recipes.filter(
+    (recipe) =>
+      recipe.name.toLowerCase().includes(searchInputValue.toLowerCase()) ||
+      recipe.ingredients.some((ingredient) =>
+        ingredient.toLowerCase().includes(searchInputValue.toLowerCase())
```
Comment (Izzibizz) [inline script.js:343]: Would have been nice to have the type accessible through the search too :)

**Review by Izzibizz — CHANGES_REQUESTED** [review]
Good work with the responsiveness, the design and all the functions working! It looks and works great :D You can be proud of this project(!)

I added a few comments about where you could shorten the code, but overall I think you made a compact nice code! I also added small comments on just the user experience of the buttons, when you search and cannot find the type and also that it is not descending on the descending button and reversed. 

Your code is easy to understand and you've added comments everywhere which helps. Looks good. You have the different sections in the right places and have good names for the functions. 

Woop wooop!

File: script.js (hunk 4 above)
Comment (almaherris) [reply script.js:325]: Fixed - thanks for noticing 🤩

File: .gitignote.md
Comment (NellaMori) [inline .gitignote.md:1]: Please add .DS_Store to your .gitignore file (if you don't have one, you need to create it) and remove the file from the repo 👍

File: recipe-images/.DS_Store
Comment (NellaMori) [inline recipe-images/.DS_Store:1]: same here 🧹

**Review by NellaMori — APPROVED** [review]
Hi Alma and Etna, GREAT JOB! You worked really really well, followed the instructions, guidelines to write good code, made the app responsive and used filters properly combining them to the genre selection. Good job with the app UI. Hat off 🎩  looking forward to seeing you continuing like this 💪

**Pull request closed without merging**